blocks/product-details/product-details.js

```diff
@@ -5,23 +5,303 @@ import { initializers } from '@dropins/tools/initializer.js';
+import ProductDescription from '@dropins/storefront-pdp/containers/ProductDescription.js';
+import ProductAttributes from '@dropins/storefront-pdp/containers/ProductAttributes.js';
+import ProductGallery from '@dropins/storefront-pdp/containers/ProductGallery.js';
 
 // Libs
 import {
-  getProduct,
   getSkuFromUrl,
   setJsonLd,
   loadErrorPage, performCatalogServiceQuery, variantsQuery,
 } from '../../scripts/commerce.js';
 import { getConfigValue } from '../../scripts/configs.js';
 import { fetchPlaceholders } from '../../scripts/aem.js';
 
+export default async function decorate(block) {
+  //  Fetch i18n labels
+  const labels = await fetchPlaceholders();
+
+  const langDefinitions = {
+    default: {
+      PDP: {
+        Product: {
+          Incrementer: { label: labels.pdpProductIncrementer },
+          OutOfStock: { label: labels.pdpProductOutofstock },
+          AddToCart: { label: labels.pdpProductAddtocart },
+          Details: { label: labels.pdpProductDetails },
+          RegularPrice: { label: labels.pdpProductRegularprice },
+          SpecialPrice: { label: labels.pdpProductSpecialprice },
+          PriceRange: {
+            From: { label: labels.pdpProductPricerangeFrom },
+            To: { label: labels.pdpProductPricerangeTo },
+          },
+          Image: { label: labels.pdpProductImage },
+        },
+        Swatches: {
+          Required: { label: labels.pdpSwatchesRequired },
+        },
+        Carousel: {
+          label: labels.pdpCarousel,
+          Next: { label: labels.pdpCarouselNext },
+          Previous: { label: labels.pdpCarouselPrevious },
+          Slide: { label: labels.pdpCarouselSlide },
+          Controls: {
+            label: labels.pdpCarouselControls,
+            Button: { label: labels.pdpCarouselControlsButton },
+          },
+        },
+        Overlay: {
+          Close: { label: labels.pdpOverlayClose },
+        },
+      },
+      Custom: {
+        AddingToCart: { label: labels.pdpCustomAddingtocart },
+      },
+    },
+  };
+
+  // Set Fetch Endpoint (Service)
+  PDP.setEndpoint(await getConfigValue('commerce-endpoint'));
+
+  // Set Fetch Headers (Service)
+  PDP.setFetchGraphQlHeaders({
+    'Content-Type': 'application/json',
+    'Magento-Environment-Id': await getConfigValue('commerce-environment-id'),
+    'Magento-Website-Code': await getConfigValue('commerce-website-code'),
+    'Magento-Store-View-Code': await getConfigValue('commerce-store-view-code'),
+    'Magento-Store-Code': await getConfigValue('commerce-store-code'),
+    'Magento-Customer-Group': await getConfigValue('commerce-customer-group'),
+    'x-api-key': await getConfigValue('commerce-x-api-key'),
+  });
+
+  // Initialize Dropins
+  initializers.register(PDP.initialize, {
+    sku: getSkuFromUrl(),
+    langDefinitions,
+    acdl: true,
+    persistURLParams: true,
+  });
+
+  // Get Initialized product data
+  const product = await PDP.getFetchedProductData();
```

Comment:
What's this? I see we are changing the prefetch logic in this PR from the scripts.js file.

Reply:
The Product Data gets now fetched earlier: the initializer instead of the render. This function gets you that data at any point.

Reply:
What do you mean earlier? It was already prefetched in `scripts.js`, which happens before any of the pdp block is run.

Reply:
You mean the custom graphql query than then is passed to the dropin that then skip it's own query. Does that sound right to you? This was patched this way to fetch data before the render which is not longer the case because the fetch happens on initialization. Want it earlier, initialization could be moved to script but I say let's test if that's even needed anymore.